ot/mpz-ot-core/src/ferret/mpcot/receiver_regular.rs

```diff
@@ -0,0 +1,192 @@
+//! MPCOT receiver for regular indices. Regular indices means the indices are evenly distributed.
+
+use mpz_core::Block;
+
+use crate::ferret::mpcot::error::ReceiverError;
+
+/// MPCOT receiver.
+#[derive(Debug, Default)]
+pub struct Receiver<T: state::State = state::Initialized> {
+    state: T,
+}
+
+impl Receiver {
+    /// Creates a new Receiver.
+    pub fn new() -> Self {
+        Receiver {
+            state: state::Initialized::default(),
+        }
+    }
+
+    /// Completes the setup phase of the protocol.
+    pub fn setup(self) -> Receiver<state::PreExtension> {
+        Receiver {
+            state: state::PreExtension { counter: 0 },
+        }
+    }
+}
+impl Receiver<state::PreExtension> {
+    /// Performs the prepare procedure in MPCOT extension.
+    /// Outputs the indices for SPCOT.
+    ///
+    /// # Arguments.
+    ///
+    /// * `alphas` - The queried indices.
+    /// * `n` - The total number of indices.
+    #[allow(clippy::type_complexity)]
+    pub fn pre_extend(
+        self,
+        alphas: &[u32],
+        n: u32,
+    ) -> Result<(Receiver<state::Extension>, Vec<(usize, u32)>), ReceiverError> {
+        let t = alphas.len() as u32;
+        if t > n {
+            return Err(ReceiverError::InvalidInput(
+                "the length of alpha should not exceed n".to_string(),
+            ));
+        }
+
+        // The range of each interval.
+        let k = (n + t - 1) / t;
+
+        let queries_length = if n % t == 0 {
+            vec![k as usize; t as usize]
+        } else {
+            let mut tmp = vec![k as usize; (t - 1) as usize];
+            tmp.push((n % k) as usize);
+            if tmp.iter().sum::<usize>() != n as usize {
+                return Err(ReceiverError::InvalidInput(
```

Comment:
Also, @xiangxiecrypto , I cant think of what values of (n,t) would cause this error to trigger. I think it never will.

Reply:
ah, nvm, (101, 10) will trigger this error